ICSharpCode.Decompiler/Ast/NameVariables.cs

```diff
@@ -61,7 +61,7 @@ public static void AssignNamesToVariables(DecompilerContext context, IEnumerable
 					// don't introduce names for variables generated by ILSpy - keep "expr"/"arg"
 					nv.AddExistingName(v.Name);
 				} else if (v.OriginalVariable != null && context.Settings.UseDebugSymbols) {
-					string varName = v.OriginalVariable.Name;
+					string varName = v.OriginalVariable.ToString();
```

Comment:
See comment above, I think it probably ought to look something like [this](https://github.com/airbreather/ILSpy/commit/4d18cd33e29de480f172b0b23aa4a1fefab845fa#diff-3ec701c5833b5402b074171880efadb4R64) using that same extension method and a `MethodDebugInformation` parameter passed in from the appropriate spot.